lib/AP.php

```diff
@@ -0,0 +1,363 @@
+use OCA\Social\Exceptions\SocialAppConfigException;
+use OCA\Social\Exceptions\UnknownItemException;
+use OCA\Social\Interfaces\Activity\AcceptInterface;
+use OCA\Social\Interfaces\Activity\AddInterface;
+use OCA\Social\Interfaces\Activity\BlockInterface;
+use OCA\Social\Interfaces\Activity\CreateInterface;
+use OCA\Social\Interfaces\Activity\DeleteInterface;
+use OCA\Social\Interfaces\Activity\FollowInterface;
+use OCA\Social\Interfaces\Activity\LikeInterface;
+use OCA\Social\Interfaces\Activity\RejectInterface;
+use OCA\Social\Interfaces\Activity\RemoveInterface;
+use OCA\Social\Interfaces\Activity\UndoInterface;
+use OCA\Social\Interfaces\Activity\UpdateInterface;
+use OCA\Social\Interfaces\Actor\PersonInterface;
+use OCA\Social\Interfaces\IActivityPubInterface;
+use OCA\Social\Interfaces\Object\NoteInterface;
+use OCA\Social\Model\ActivityPub\ACore;
+use OCA\Social\Model\ActivityPub\Activity\Accept;
+use OCA\Social\Model\ActivityPub\Activity\Add;
+use OCA\Social\Model\ActivityPub\Activity\Block;
+use OCA\Social\Model\ActivityPub\Activity\Create;
+use OCA\Social\Model\ActivityPub\Activity\Delete;
+use OCA\Social\Model\ActivityPub\Activity\Follow;
+use OCA\Social\Model\ActivityPub\Activity\Like;
+use OCA\Social\Model\ActivityPub\Activity\Reject;
+use OCA\Social\Model\ActivityPub\Activity\Remove;
+use OCA\Social\Model\ActivityPub\Activity\Undo;
+use OCA\Social\Model\ActivityPub\Activity\Update;
+use OCA\Social\Model\ActivityPub\Actor\Person;
+use OCA\Social\Model\ActivityPub\Object\Document;
+use OCA\Social\Model\ActivityPub\Object\Image;
+use OCA\Social\Model\ActivityPub\Object\Note;
+use OCA\Social\Model\ActivityPub\Object\Tombstone;
+use OCA\Social\Service\ConfigService;
+use OCP\AppFramework\QueryException;
+
+
+/**
+ * Class AP
+ *
+ * @package OCA\Social
+ */
+class AP {
+
+
+	use TArrayTools;
+
+
+	const REDUNDANCY_LIMIT = 10;
+
+
+	/** @var AcceptInterface */
+	public $acceptInterface;
+
+	/** @var AddInterface */
+	public $addInterface;
+
+	/** @var BlockInterface */
+	public $blockInterface;
+
+	/** @var CreateInterface */
+	public $createInterface;
+
+	/** @var DeleteInterface */
+	public $deleteInterface;
+
+	/** @var FollowInterface */
+	public $followInterface;
+
+	/** @var LikeInterface */
+	public $likeInterface;
+
+	/** @var PersonInterface */
+	public $personInterface;
+
+	/** @var NoteInterface */
+	public $noteInterface;
+
+	/** @var RejectInterface */
+	public $rejectInterface;
+
+	/** @var RemoveInterface */
+	public $removeInterface;
+
+	/** @var UndoInterface */
+	public $undoInterface;
+
+	/** @var UpdateInterface */
+	public $updateInterface;
+
+	/** @var ConfigService */
+	public $configService;
+
+
+	/** @var AP */
+	public static $activityPub = null;
+
+
+	/**
+	 * AP constructor.
+	 */
+	public function __construct() {
+	}
+
+
+	/**
+	 *
+	 */
+	public static function init() {
+		$ap = new AP();
+		try {
+			$ap->acceptInterface = \OC::$server->query(AcceptInterface::class);
```

Comment:
Can't we use proper dependency injection here?

Reply:
This is the only solution I found to avoid cross-injecting of the Parser (including all Interface, and their service on complex activitypub objects. The parser needs to be a static object, and it makes all the parsing and the creation of an ActivityPub object a lot easier.